tests/singlefolder.py

```diff
@@ -153,7 +153,7 @@ def _put_file(
     ):
         with self.Map(self._path, self._lock) as m:
 
-            if dst_path not in m:
+            if dst_path not in m or checksum != m[dst_path][1]:
```

Comment:
Why do we need this extra check here?

Reply:
Because we only want to skip when the checksums match. Otherwise the file has changed and we need to replace it. Was a bug before.